src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -109,6 +110,10 @@ export class DeviceSourceManager implements IDisposable {
      * @returns DeviceSource object
      */
     public getDeviceSource<T extends DeviceType>(deviceType: T, deviceSlot?: number): Nullable<DeviceSource<T>> {
+        if (!this._checkConnectedGamepads) {

```

Comment:
Why do we want to put this behavior behind a flag (defaulted false)?

Reply:
The reason why I set this behind a flag is that I only want this to be run once and only when the first GetDeviceSource call happens.  All that this will do is just catch already connected gamepads that would otherwise miss being registered because they never throw a gamepadconnected event.

Reply:
If we only want this to happen once, and to catch already connected devices before our DSM was created, then why not do this at construction? Also, I'm not sure if I see any logic that ensures that this is only called once, and only on the first time this function is called...

Also, maybe invert the condition logic? it seems like you're checking for the opposite of what your variable name `DeviceSourceManager._checkConnectedGamepads` implies.

Reply:
I can fix the variable naming.  As for why I have it here, if I run it in the constructor and the user adds any functions to the Connected/Disconnected observables, those custom observable functions would never run because all of the registration would take place before the user had a chance to add them to the observables.  By putting them in the getDeviceSource, I will get already connected gamepads to be registered and activate those functions.

Reply:
Just an update, after talking with Gary and testing some things, I'll be putting this call in the constructor.  The original idea for my choice was that I felt like custom callbacks should happen before devices are added to the DSM.  I wanted the users to be able to handle controller context switching when a device is connected (eg. showing Xbox controls when an xbox contorller is connected) but looking at it now, I don't think that this would properly support that scenario.  TBH, we should probably leave that to the users and keep things simple.